src/histolab/filters/morphological_filters_functional.py

```diff
@@ -60,3 +63,32 @@ def remove_small_objects(
             np_mask, new_min_size, avoid_overmask, overmask_thresh
         )
     return mask_no_small_object
+
+
+def watershed_segmentation(np_mask: np.ndarray) -> np.ndarray:
+    """Segment and label an binary mask with Watershed segmentation [1]_
+
+    The watershed algorithm treats pixels values as a local topography (elevation).
+
+    Parameters
+    ----------
+    np_mask : np.ndarray
+        Input mask
+
```

Comment:
region_shape: int
the local region within which to search for image peaks is  defined as a squared area region_shape x region_shape